nfs.go

```diff
@@ -49,14 +59,40 @@ var zeroHandle handle
 var statusHandle = handle{0: 'S', 1: 'T', 2: 'A', 3: 'T', 4: 'U', 5: 'S'}
 
 type NFSHandler struct {
-	fs *FS
-
+	fs          *FS
 	nfs.Handler // temporary embedding during dev to watch what panics
 
-	mu          sync.Mutex
-	handle      map[handle]handleTarget
-	readCache   map[handle]*readCacheEntry
+	metricFileContentCacheHit  expvar.Int
+	metricFileContentCacheMiss expvar.Int
+	metricFileContentCacheFill expvar.Int
+
+	mu sync.Mutex // protects all the following fields
+
 	statusCache *statusMeta
+
+	// handle maps from an NFS handle to the path it corresponds to.
+	//
+	// TODO(bradfitz): LRU-ify this. this was once load bearing but it's now a
+	// cache now that all handles can map back to a handleTarget anyway
+	handle map[handle]handleTarget
+
+	entCache  lru.Cache[handle, *readCacheEntry]
+	blobCache lru.Cache[blobHash, []byte]
+	blobCount map[blobHash]int // ref count of blobHash in entCache
+}
+
+func (n *NFSHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	// TODO(bradfitz): add a proper Prometheus handler. These are temporary as
+	// of 2025-08-22 while tuning and deciding what metrics we actually want.
+
+	w.Header().Set("Content-Type", "text/plain")
+	fmt.Fprintf(w, "handles = %v\n", len(n.handle))
+	fmt.Fprintf(w, "entCache = %v\n", n.entCache.Len())
+	fmt.Fprintf(w, "blobCache = %v ents, %v size\n", n.blobCache.Len(), n.blobCache.Size())
```

Comment:
would be nice to display the size with units, maybe as MB/GB?
<img width="480" height="165" alt="nfs" src="https://github.com/user-attachments/assets/da259e2d-f31c-477e-b5f8-763c3d26ed28" />

Reply:
deleted. now prometheus metrics instead.